packages/did/src/DidDetails/FullDidDetails.ts

```diff
@@ -96,7 +100,11 @@ export function getKeysForExtrinsic(
  * @returns The next valid nonce, i.e., the nonce currently stored on the blockchain + 1, wrapping around the max value when reached.
  */
 export async function getNextNonce(did: DidDocument): Promise<BN> {
-  const currentNonce = await queryNonce(did.uri)
+  const api = ConfigService.get('api')
+  const queried = await api.query.did.did(didToChain(did.uri))
+  const currentNonce = queried.isSome
+    ? didFromChain(queried).lastTxCounter
```

Comment:
i don't think it makes a lot of sense to decode the entire struct to just access one field on it

Reply:
It’s not really a performance hit and reduces cognitive load by reusing the building blocks